public/app/features/alerting/unified/components/rule-editor/TemplateStep/TemplateStep.tsx

```diff
@@ -0,0 +1,236 @@
+    getData();
+  }, [getData]);
+
+  return (
+    <RuleEditorSection stepNo={2} title="Template details">
+      <Field
+        label={Messages.templateField}
+        description={Messages.tooltips.template}
+        error={errors.template?.message}
+        invalid={!!errors.template?.message}
+      >
+        <Controller
+          name="template"
+          rules={{ required: { value: true, message: Messages.errors.template } }}
+          render={({ field: { value, onChange } }) => (
+            <Select
+              id="template"
+              isLoading={loadingTemplates}
+              disabled={loadingTemplates}
+              placeholder={loadingTemplates ? Messages.loadingTemplates : undefined}
+              value={templateOptions?.find((opt) => opt.value?.name === value?.name)}
+              onChange={(selectedTemplate) => handleTemplateChange(selectedTemplate.value, onChange)}
+              options={templateOptions}
+              data-testid="template-select-input"
+            />
+          )}
+        />
+      </Field>
+      <Field
+        label={Messages.nameField}
+        description={Messages.tooltips.name}
+        error={errors.name?.message}
+        invalid={!!errors.name?.message}
+      >
+        <Input id="name" {...register('name', { required: { value: true, message: Messages.errors.name } })} />
+      </Field>
+
+      {/* TODO add remaining params as API starts supporting them
+      https://github.com/percona/pmm-managed/blob/PMM-2.0/models/template_model.go#L112 */}
+      {currentTemplate?.params?.map(
+        ({ float, type, name, summary, unit }) =>
+          type === TemplateParamType.FLOAT && (
+            <Field
+              key={name}
+              label={`${name[0].toUpperCase()}${name.slice(1)}`}
+              description={Messages.getFloatDescription(summary, unit, float)}
+              // @ts-ignore
+              error={errors[name]?.message}
+              // @ts-ignore
+              invalid={!!errors[name]?.message}
+            >
+              <Input
+                type="number"
+                // @ts-ignore
+                {...register(name, {
+                  required: { value: true, message: Messages.errors.floatParamRequired(name) },
+                  min: float?.hasMin
+                    ? { value: float.min || 0, message: Messages.errors.floatParamMin(float.min || 0) }
+                    : undefined,
+                  max: float?.hasMax
+                    ? { value: float.max || 0, message: Messages.errors.floatParamMax(float.max || 0) }
+                    : undefined,
+                })}
+                name={name}
+                defaultValue={`${float?.default}`}
+              />
+            </Field>
+          )
+      )}
+
+      <Field
+        label={Messages.durationField}
+        description={Messages.tooltips.duration}
+        error={errors.duration?.message}
+        invalid={!!errors.duration?.message}
+      >
+        <Input
```

Comment:
why don't we use textinputfield and ready-made select here from platform/core? are they not suitable for us?

Reply:
Yeah, the thing is that they use a different type of form. Not `react-final-form`, as we've been using, but `react-hook-form`. It's a bit different, and out components have two problems now:

1. They were made to work with `react-final-form`;
2. They look a lot different than Grafana's form fields, which is bad for users (different UI/UX).

WIth this is mind and given we want to look more and more like Grafana's native pages, I'm inclined towards using their form fields from now on as much as possible.

Reply:
Okey, got it)